### PR TITLE
Updated layout for listing page

### PR DESCRIPTION
**Pull request by Lukewh**
Fixes https://github.com/canonical-websites/snapcraft.io/issues/1705

## QA

- Pull the branch
- `./run`
- Visit http://0.0.0.0:8004/<snap>/listing
or
- Use the demo
- See the new layout - ensure fields and combinations still work as expected
- :+1:

### CODE REVIEW

File: templates/publisher/listing.html [hunk 1]
```diff
@@ -162,18 +144,18 @@
             <select
               name="primary_category"
               {% if snap_categories.categories[0] %}
-                value="{{ snap_categories.categories[0].name }}"
+              value="{{ snap_categories.categories[0].name }}"
```
Comment (bartaz) [inline templates/publisher/listing.html]: From code readability point of view I think indent in the template code (between `{% if %}`, `{% for %}`, etc is more important than indent in rendered HTML.

File: templates/publisher/listing.html [hunk 2]
```diff
@@ -439,66 +449,35 @@
                 name="public_metrics_territories"
                 id="public-metrics-territories"
                 {% if not contains(public_metrics_blacklist, 'installed_base_by_country_percent') %}
-                checked="checked"
-                {% endif %}
-                {% if not public_metrics_enabled %}
-                disabled="disabled"
-                {% endif %}
+              checked="checked"
```
Comment (bartaz) [inline templates/publisher/listing.html]: Something strange happened to indent there (and couple similar sections below)

**Comment by bartaz** [comment]
Headings with `p-heading--five` seem quite small.
I feel like `p-heading--four` would work a bit better:

Screenshot 2019-03-25 at 11 43 33

File: templates/publisher/listing.html (hunk 1 above)
Comment (Lukewh) [reply templates/publisher/listing.html]: Ah thanks, I think my editor did some tomfoolery when moving things around

**Comment by hughesjordan** [comment]
@Lukewh Can we make it so that the remove icons are offset by the same distance from the inputs?
Screenshot 2019-03-26 at 10 00 46 am

Also, when on mobile there are a few issues. The remove 'snap icon' icon is now to the left and the remove 'category' icon underneath as it's in another column. Maybe something to save and tackle all mobile issues at once.
Screenshot 2019-03-26 at 10 31 27 am

Last thing would be spacing, if you click on the image below you can see a comparison between how it is currently and a suggestion.

The suggestion (image on the right) has a little more spacing between inputs to give them a little more breathing room and then more generous spacing between the `icon/title/category`, `video/images` and `summary/description` areas. This is to group the relevant areas here as they would be presented on the snap listing page.
Comparison

**Pull request merged**